jsonrpc/server.go

```diff
@@ -399,6 +408,10 @@ func (s *Server) buildArguments(ctx context.Context, params any, method Method)
 	case reflect.Map:
 		paramsMap := params.(map[string]any)
 
+		if len(paramsMap) == 0 && allParamsAreOptional {
```

Comment:
I dont think this is needed, the loop below will construct a list with the default argument anyways.

Reply:
Yep, you're right - passing {} works the same with/without this changes. Will remove